### PR TITLE
Adds ingress and cert-manager

### PR DESCRIPTION
**Pull request by lypht**
Signed-off-by: Josh Hull <josh.hull@lypht.com>

### CODE REVIEW

File: helm/java-prototype/values.yaml [hunk 1]
```diff
@@ -25,18 +25,8 @@ service:
   type: NodePort
   port: 8080
 
-ingress:
-  enabled: false
-  annotations: {}
```
Comment (paulczar) [inline helm/java-prototype/values.yaml]: We should allow user to specify custom annotations,  not sure why you're removing this ?

File: helm/java-prototype/values.yaml [hunk 2]
```diff
@@ -25,18 +25,8 @@ service:
...
-    # kubernetes.io/ingress.class: nginx
-    # kubernetes.io/tls-acme: "true"
-  path: /
-  hosts:
-    - chart-example.local
-  tls: []
```
Comment (paulczar) [inline helm/java-prototype/values.yaml]: We should allow user to specify custom TLS,  not sure why you're removing this ?

File: helm/java-prototype/values.yaml [hunk 3]
```diff
@@ -25,18 +25,8 @@ service:
...
-  #  - secretName: chart-example-tls
-  #    hosts:
-  #      - chart-example.local
+basedomain: heptio.com
```
Comment (paulczar) [inline helm/java-prototype/values.yaml]: can you add comments above to show where these are being used and why ?

File: helm/java-prototype/values.yaml (hunk 2 above)
Comment (lypht) [reply helm/java-prototype/values.yaml]: Nginx TLS notation previously caused kube-lego to choke.  Note sure if this has been resolved in `cert-manager`; more in the troubleshooting readme for `contour`: https://github.com/heptio/contour/blob/695240e1f8033b83eafc39a7cb202a1c66e65b35/docs/troubleshooting.md

Comment (paulczar) [reply helm/java-prototype/values.yaml]: sure, that's why it should be configurable via `tls: []` , If I'm using minikube to test I want to be able to use nginx ingress, if I'm using GKE I want to be able to use their ingress. I'm happy for defaults to be empty, but want to be able to easily slip my own settings is via `values.yaml`.

**Pull request merged**